File: 1021.cpp
//1021.身份验证
#include<iostream>
#include<vector>
#include<string>

using namespace std;

int main()
{
    int n;
    cin>>n;
    vector <string> s(n);
    vector <int> sum(n);
    int Z[17]={7,9,10,5,8,4,2,1,6,3,7,9,10,5,8,4,2};
    char M[11]={'1','0','X','9','8','7','6','5','4','3','2'};
    for(int i=0;i<n;i++)
    {
        cin>>s[i];
    }
    int t=0;
    for(int i=0;i<n;i++)
    {
        sum[i]=0;
        for(int j=0;j<17;j++)
        {
            if(!isdigit(s[i][j]))
            {
                sum[i]=-1;
                break;
            }
            sum[i]=sum[i]+Z[j]*(s[i][j]-'0');
        }
        sum[i]=sum[i]%11;
        if(M[sum[i]]!=s[i][17])
        {
            for(int j=0;j<18;j++) 
                cout<<s[i][j];
            cout<<endl;
            t++;
        }
    }
    if(t==0) cout<<"All passed"<<endl;
    return 0;
}

File: 1021.py
#!/usr/bin/env python3
#未测试
if __name__ == "__main__":
    line = int(input())
    IDcodeList = list()
    while line:
        IDcode = input()
        IDcodeList.append(IDcode)
        line -= 1
    weight_list = [7，9，10，5，8，4，2，1，6，3，7，9，10，5，8，4，2]
    sum_lst = list()
    for i, IDcode in enumerate(IDcodeList):
        sum = 0
        for char in IDcode:
            if not char.isdigit():
                sum = -1
                break
            sum += weight_list[i] * int(char)
        sum_list.append(sum)
    Z_lst = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    #M: 1 0 X 9 8 7 6 5 4 3 2
    M_lst = ['1', '0', 'X', '9', '8', '7', '6', '5', '3', '2', '1']
    ZM_dict = {0:'1', 1:'0', 2:'X', 3:'9', 4:'8', 5:'7', 6:'5', 7:'3', 8:'2', 9:'2', 10:'1'}
    afterMod_lst = [(i % 11) for i in sum_lst]
    valid_IDcode = list()
    for i ,afterModNum in enumerate(afterMod_lst):
        if ZM_dict[afterModNum] == IDcodeList[i][-1]:
            print(IDcodeList[i])
    



